src/pages/OnboardingAccounting/BaseOnboardingAccounting.tsx

```diff
@@ -112,13 +126,44 @@ function BaseOnboardingAccounting({shouldUseNativeStyles}: BaseOnboardingAccount
                 text={translate('common.confirm')}
                 // eslint-disable-next-line rulesdir/prefer-early-return
                 onPress={() => {
-                    if (!userReportedIntegration) {
+                    if (userReportedIntegration === undefined) {
                         setError(translate('onboarding.purpose.errorSelection'));
                         // eslint-disable-next-line no-useless-return
                         return;
                     }
 
-                    // TODO: call CompleOnboarding API after the back-end PR is complete
+                    if (!onboardingPurposeSelected) {
+                        return;
+                    }
```

Comment:
In which case we reach this step and with no purpose selected?

Reply:
We will not reach this case, but it's used to prevent the ts error. The same way is used here.

https://github.com/Expensify/App/blob/8d1972651eefdd89a01cbccc1b0ed0b7240ac9ed/src/pages/OnboardingPersonalDetails/BaseOnboardingPersonalDetails.tsx#L53